src/app/base/components/Header/Header.test.tsx

```diff
@@ -218,6 +223,33 @@ it("highlights sub-urls", () => {
   expect(currentMenuItem).toHaveTextContent("Machines");
 });
 
+it("displays a warning icon next to controllers if vault is not fully configured", () => {
+  state.controller.items = [
+    controllerFactory({ vault_configured: true }),
+    controllerFactory({ vault_configured: false }),
+  ];
+  renderWithBrowserRouter(<Header />, { route: "/", wrapperProps: { state } });
+
+  const controllerLink = screen.getByRole("link", { name: "Controllers" });
+  const warningIcon = within(controllerLink).getByTestId("warning-icon");
```

Comment:
Normally I wouldn't use a test id, but using an aria label on the icon messes with how RTL reads the name of the button in the header, and this is a reliable way of seeing if this icon is rendering in the document.